drivers/Watchdog.cpp

```diff
@@ -0,0 +1,110 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifdef DEVICE_WATCHDOG
+
+#include "Watchdog.h"
+
+namespace mbed {
+
+Watchdog *Watchdog::_first = NULL;
+
+Watchdog::Watchdog(uint32_t timeout, const char *const str): _name(str)
```

Comment:
Why is one of the attributes initialised using the member initialiser list and the rest in the constructor body?

Reply:
Make sure str keep as constant while referred by _name pointer. and keep the other member initialization in definition to make more readability of the code.